egs/librispeech/ASR/conformer_ctc/ali.py

```diff
@@ -223,21 +228,32 @@ def compute_alignments(
             use_double_scores=params.use_double_scores,
         )
 
-        ali_ids = get_alignments(best_path)
-        assert len(ali_ids) == len(cut_list)
-        for cut, ali in zip(cut_list, ali_ids):
-
-            cut.token_alignment = writer.store_array(
+        labels_ali = get_alignments(best_path, kind="labels")
+        aux_labels_ali = get_alignments(best_path, kind="aux_labels")
```

Comment:
Does "labels" stand for BPEs/phones and "aux_labels" for words? If yes then it might be clearer to name these attrs for what they represent.

Reply:
It uses a CTC topology for decoding here. Labels are BPE tokens or phones, while aux labels are also BPE tokens or phones but without repeat.

For instance, if labels are

c c blk blk a a a blk t t blk

then aux labels are

c blk blk blk a blk blk blk t blk blk


The meaning of aux labels depends on the kind of FSA you are using.

If you are using a CTC graph or CTC topo, the aux labels are usually BPE tokens or phones.

If you are using an HLG graph, then the aux labels are usually words.

Reply:
Gotcha, makes sense.

Reply:
The framewise alignment using BPE tokens without repeat can be used together with the embedding stuff to compute auxiliary loss(es).